circuits/circom/utils/verifySignature.circom

```diff
@@ -0,0 +1,120 @@
+pragma circom 2.0.0;
+
+// circomlib imports
+include "./compconstant.circom";
+include "./comparators.circom";
+include "./pointbits.circom";
+include "./bitify.circom";
+include "./escalarmulany.circom";
+include "./escalarmulfix.circom";
+// local imports
+include "./hashers.circom";
+
+/**
+ * Variant of the EdDSAPoseidonVerifier template from circomlib
+ * https://github.com/iden3/circomlib/blob/master/circuits/eddsa.circom
+ */
+template EdDSAPoseidonVerifier_patched() {
+    // The x and y coordinates of the public key.
+    signal input Ax;
+    signal input Ay;
+    // Signature scalar.
+    signal input S;
+    // The x and y coordinates of the signature point.
+    signal input R8x;
+    signal input R8y;
+    // Message hash.
+    signal input M;
+
+    signal output valid;
+
+    // Ensure S<Subgroup Order.
+    // convert the signature scalar S into its binary representation.
+    var computedNum2Bits[254] = Num2Bits(254)(S);
+
+    var computedCompConstantIn[254];
+    for (var i=0; i<253; i++) {
+        computedCompConstantIn[i] = computedNum2Bits[i];
+    }
+    computedCompConstantIn[253] = 0;
+
+    // A component that ensures S is within a valid range, 
+    // comparing it against a constant representing the subgroup order.
+    var computedCompConstant = CompConstant(2736030358979909402780800718157159386076813972158567259200215660948447373040)(computedCompConstantIn);
+
+    // Calculate the h = H(R,A, msg).
+    var computedH2Bits[254] = Num2Bits_strict()(PoseidonHasher(5)([R8x, R8y, Ax, Ay, M]));
+
+    // These components perform point doubling operations on the public key
+    // to align it within the correct subgroup as part of the verification process.
+    var (computedDbl1XOut, computedDbl1YOut) = BabyDbl()(Ax, Ay);
+    var (computedDbl2XOut, computedDbl2YOut) = BabyDbl()(computedDbl1XOut, computedDbl1YOut);
+    var (computedDbl3XOut, computedDbl3YOut) = BabyDbl()(computedDbl2XOut, computedDbl2YOut);
+
+    // A component that performs scalar multiplication of the 
+    // adjusted public key by the hash output, essential for the verification calculation.
+    var computedEscalarMulAny[2] = EscalarMulAny(254)(computedH2Bits, [computedDbl3XOut, computedDbl3YOut]);
+
+    // Compute the right side: right =  R8 + right2.
+    var (computedAddRightXOut, computedAddRightYOut) = BabyAdd()(R8x, R8y, computedEscalarMulAny[0], computedEscalarMulAny[1]);
+
+    // Calculate left side of equation left = S*B8.
```

Comment:
really tiny pick: should we unify the equation? e.g. `left = S * R8`, `right = R8 + right2` or `left=S*R8`, `right=R8+right2`

Reply:
Let's stick with the former!